feat: calculate fields based on ageInMonths changes

MatiasArriola opened the pull request:
**Task:** https://app.clickup.com/t/869aedje0

**Implementation:**

- Clear `ageInMonths` if age > 5
- When `ageInMonths` changes, update `dateOfBirth` to the first day of the corresponding month and update `age`
- Added extra conditions to avoid unnecessary chained updates

MatiasArriola committed:
feat: calculate fields based on ageInMonths changes

Clear ageInMonths if age > 5
Handle chained updates

Ramon-Jimenez reviewed (approved):
All good. I am not sure about clearing Months if Age > 5, I'll leave the client to test this also with the Program Rules associated to hiding before merging this one.

MatiasArriola committed:
fix: changing ageInMonths that triggers year change

dob is not overriden anymore with the age calculation if ageInMonths is set

Ramon-Jimenez reviewed (changes requested):
@MatiasArriola it is now working OK but there is a usecase that is behaving different from the current Production plugin. It allows you to edit Age or AgeInMonths if DoBIsKnown=true, and it allows you to edit DoB if DoBIsKnown=false. This is currently not possible in Production (you can edit the value but as soon as you click elsewhere the calculation kicks in again and resets the value to the calculated one) and it is actually the expected behavior, as the user should not be able to edit those fields depending on the DoBIsKnown value.

MatiasArriola committed:
fix: revert back age or ageInMonths if dobKnown=true, revert dob if dobKnown=false

change individual useSetField to useSetFields hook, track internal state using a ref to distinguish from props coming from Capture.
change useEffect/useCallback dependencies
